File: Data Structures & Algorithms/subarray-sum-equals-k/submission-3.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        int n = nums.length;
        int[] sumArr = new int[n+1];
        for(int i=0;i<n;i++){
            sumArr[i+1] = sumArr[i]+nums[i];
        }
        int count=0;
        for(int i=1;i<=n;i++){
            for(int j=i;j<=n;j++){
                if(sumArr[j]-sumArr[j-i]==k)
                    count++;
            }
        }
        return count;
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-3.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map<char,int> m;
        int i=0;
        int j=0;
        int n = s.size();
        int max_window = 0;
        int count=0;
        while(j<n){
            if(m.find(s[j])!=m.end()){
                while(i<=m[s[j]]){
                    m.erase(s[i++]);
                }
            }
            m[s[j]]=j;
            max_window = max(j - i + 1,max_window);
            j++;
        }
        return max_window;
    }
};
